manager/controlapi/service.go

```diff
@@ -472,18 +472,36 @@ func (s *Server) validateNetworks(networks []*api.NetworkAttachmentConfig) error
 	return nil
 }
 
-func validateMode(s *api.ServiceSpec) error {
+// validateMode returns two values: the first is a bool indicating whether the
+// service is a job-type service or not, and the second is an error resulting
+// from a failure of mode validation.
+//
+// the returned bool is used because validation of certain service properties
+// is different if a service is in job mode.
```

Comment:
Honestly, it seems a bit contrived to me to have this function do this too, which is quite different from its primary purpose; makes the code a bit more confusing and harder to follow.

Instead, how about having a
```
func isJobSpec(spec *api.ServiceSpec) bool {
	mode := spec.GetMode()
	_, isGlobalJob := mode.(*api.ServiceSpec_GlobalJob)
	if isGlobalJob {
		return true
	}
	_, isReplicatedJob := mode.(*api.ServiceSpec_ReplicatedJob)
	return isReplicatedJob
}
```
helper (which could arguably even be a member method on `api.ServiceSpec`s, and not having `validateMode` do two very different things?

Besides, `isJobSpec` could be reused below (lines 721-722)